In [9]:
from nltk.tokenize import word_tokenize
from nltk.corpus import stopwords

``A corpus is a collection of authentic texts or audio that are organized into datasets and representative of a specific language or language variety``

In [10]:
corpus = "India, officially the Republic of India (ISO: Bhārat Gaṇarājya),is a country in South Asia. It is the seventh-largest country by area; the most populous country as of June 2023;and from the time of its independence in 1947, the world's most populous democracy.Bounded by the Indian Ocean on the south, the Arabian Sea on the southwest, and the Bay of Bengal on the southeast, it shares land borders with Pakistan to the west;China, Nepal, and Bhutan to the north; and Bangladesh and Myanmar to the east. In the Indian Ocean, India is in the vicinity of Sri Lanka and the Maldives; its Andaman and Nicobar Islands share a maritime border with Thailand, Myanmar, and Indonesia."

In [11]:
corpus

"India, officially the Republic of India (ISO: Bhārat Gaṇarājya),is a country in South Asia. It is the seventh-largest country by area; the most populous country as of June 2023;and from the time of its independence in 1947, the world's most populous democracy.Bounded by the Indian Ocean on the south, the Arabian Sea on the southwest, and the Bay of Bengal on the southeast, it shares land borders with Pakistan to the west;China, Nepal, and Bhutan to the north; and Bangladesh and Myanmar to the east. In the Indian Ocean, India is in the vicinity of Sri Lanka and the Maldives; its Andaman and Nicobar Islands share a maritime border with Thailand, Myanmar, and Indonesia."

In [12]:
word_tokenize(corpus)

['India',
 ',',
 'officially',
 'the',
 'Republic',
 'of',
 'India',
 '(',
 'ISO',
 ':',
 'Bhārat',
 'Gaṇarājya',
 ')',
 ',',
 'is',
 'a',
 'country',
 'in',
 'South',
 'Asia',
 '.',
 'It',
 'is',
 'the',
 'seventh-largest',
 'country',
 'by',
 'area',
 ';',
 'the',
 'most',
 'populous',
 'country',
 'as',
 'of',
 'June',
 '2023',
 ';',
 'and',
 'from',
 'the',
 'time',
 'of',
 'its',
 'independence',
 'in',
 '1947',
 ',',
 'the',
 'world',
 "'s",
 'most',
 'populous',
 'democracy.Bounded',
 'by',
 'the',
 'Indian',
 'Ocean',
 'on',
 'the',
 'south',
 ',',
 'the',
 'Arabian',
 'Sea',
 'on',
 'the',
 'southwest',
 ',',
 'and',
 'the',
 'Bay',
 'of',
 'Bengal',
 'on',
 'the',
 'southeast',
 ',',
 'it',
 'shares',
 'land',
 'borders',
 'with',
 'Pakistan',
 'to',
 'the',
 'west',
 ';',
 'China',
 ',',
 'Nepal',
 ',',
 'and',
 'Bhutan',
 'to',
 'the',
 'north',
 ';',
 'and',
 'Bangladesh',
 'and',
 'Myanmar',
 'to',
 'the',
 'east',
 '.',
 'In',
 'the',
 'Indian',
 'Ocean',
 ',',
 'India',

In [13]:
len(word_tokenize(corpus))

140

`` So that the word don't repeat and sound similar like the and The, therefore we make all the text in lowercase.``

``Vocabulary Size : The set of unique words used in the text corpus ,Greater the Vocabualry size more diffficult it's to train the model.``

In [16]:
# Removing stopwords
l = []
for word in word_tokenize(corpus):
    if(word.lower() not in stopwords.words('english')):
        l.append(word)
print(l)

['India', ',', 'officially', 'Republic', 'India', '(', 'ISO', ':', 'Bhārat', 'Gaṇarājya', ')', ',', 'country', 'South', 'Asia', '.', 'seventh-largest', 'country', 'area', ';', 'populous', 'country', 'June', '2023', ';', 'time', 'independence', '1947', ',', 'world', "'s", 'populous', 'democracy.Bounded', 'Indian', 'Ocean', 'south', ',', 'Arabian', 'Sea', 'southwest', ',', 'Bay', 'Bengal', 'southeast', ',', 'shares', 'land', 'borders', 'Pakistan', 'west', ';', 'China', ',', 'Nepal', ',', 'Bhutan', 'north', ';', 'Bangladesh', 'Myanmar', 'east', '.', 'Indian', 'Ocean', ',', 'India', 'vicinity', 'Sri', 'Lanka', 'Maldives', ';', 'Andaman', 'Nicobar', 'Islands', 'share', 'maritime', 'border', 'Thailand', ',', 'Myanmar', ',', 'Indonesia', '.']


In [17]:
len(l)

83

In [27]:
# Removing punctuations
k = []
for word in word_tokenize(corpus):
    if(word.lower() not in stopwords.words('english')) and (len(word) >= 2):
        k.append(word.lower())
print(k)

['india', 'officially', 'republic', 'india', 'iso', 'bhārat', 'gaṇarājya', 'country', 'south', 'asia', 'seventh-largest', 'country', 'area', 'populous', 'country', 'june', '2023', 'time', 'independence', '1947', 'world', "'s", 'populous', 'democracy.bounded', 'indian', 'ocean', 'south', 'arabian', 'sea', 'southwest', 'bay', 'bengal', 'southeast', 'shares', 'land', 'borders', 'pakistan', 'west', 'china', 'nepal', 'bhutan', 'north', 'bangladesh', 'myanmar', 'east', 'indian', 'ocean', 'india', 'vicinity', 'sri', 'lanka', 'maldives', 'andaman', 'nicobar', 'islands', 'share', 'maritime', 'border', 'thailand', 'myanmar', 'indonesia']


In [28]:
#to get most unique and useful words, without repetition
len(set(k))

52

`` 140 words made to 52 after removal of stopwrods, punctutation, and repititions.``

In [29]:
vocab = list(set(k))

In [30]:
vocab

['democracy.bounded',
 'june',
 'shares',
 'south',
 'north',
 'bangladesh',
 'borders',
 'vicinity',
 'bhārat',
 'land',
 'bhutan',
 'time',
 'east',
 'maritime',
 'bengal',
 'iso',
 'indian',
 'area',
 'world',
 'india',
 "'s",
 'thailand',
 'southeast',
 'andaman',
 'islands',
 '1947',
 'gaṇarājya',
 'country',
 'west',
 'china',
 'sri',
 'maldives',
 'sea',
 'arabian',
 'independence',
 'pakistan',
 'indonesia',
 'populous',
 'border',
 'ocean',
 'bay',
 'nepal',
 'nicobar',
 'seventh-largest',
 'southwest',
 'share',
 'republic',
 'officially',
 'asia',
 '2023',
 'myanmar',
 'lanka']

`` Performing Sentence tokenization.``

In [31]:
from nltk.tokenize import sent_tokenize

In [34]:
for sent in sent_tokenize(corpus):
    print(sent)

LookupError: 
**********************************************************************
  Resource punkt not found.
  Please use the NLTK Downloader to obtain the resource:

  >>> import nltk
  >>> nltk.download('punkt')
  
  For more information see: https://www.nltk.org/data.html

  Attempted to load tokenizers/punkt/675.pickle

  Searched in:
    - 'C:\\Users\\LENOVO/nltk_data'
    - 'c:\\Users\\LENOVO\\AppData\\Local\\Programs\\Python\\Python311\\nltk_data'
    - 'c:\\Users\\LENOVO\\AppData\\Local\\Programs\\Python\\Python311\\share\\nltk_data'
    - 'c:\\Users\\LENOVO\\AppData\\Local\\Programs\\Python\\Python311\\lib\\nltk_data'
    - 'C:\\Users\\LENOVO\\AppData\\Roaming\\nltk_data'
    - 'C:\\nltk_data'
    - 'D:\\nltk_data'
    - 'E:\\nltk_data'
    - ''
**********************************************************************
